bundles/org.openhab.core/src/test/java/org/openhab/core/common/ThreadPoolManagerTest.java

```diff
@@ -35,6 +42,135 @@
 @NonNullByDefault
 public class ThreadPoolManagerTest {
 
+    @BeforeAll
+    public static void enableSequentialScheduledExecutorService() {
+        System.setProperty("org.openhab.core.common.SequentialScheduledExecutorService", "true");
+    }
+
+    @Test
+    public void testSequentialExecutorServiceAssumptions() {
+        Callable<Boolean> callable = () -> true;
+        Runnable runnable = () -> {
+        };
+
+        ScheduledExecutorService service = Executors.newScheduledThreadPool(1);
+
+        assertTrue(service.submit(runnable) instanceof RunnableFuture);
+        assertTrue(service.submit(callable) instanceof RunnableFuture);
+
+        assertTrue(service.schedule(runnable, 1, TimeUnit.SECONDS) instanceof RunnableFuture);
+        assertTrue(service.schedule(callable, 1, TimeUnit.SECONDS) instanceof RunnableFuture);
+
+        var fixedRate = service.scheduleAtFixedRate(runnable, 1, 1, TimeUnit.SECONDS);
+
+        try {
+            assertTrue(fixedRate instanceof RunnableFuture);
+        } finally {
+            fixedRate.cancel(false);
+        }
+
+        var fixedDelay = service.scheduleWithFixedDelay(runnable, 1, 1, TimeUnit.SECONDS);
+
+        try {
+            assertTrue(fixedDelay instanceof RunnableFuture);
+        } finally {
+            fixedDelay.cancel(false);
+        }
+    }
+
+    @Test
+    public void testExecutionIsSequentialInSequentialExecutorService() {
+        ScheduledExecutorService service = ThreadPoolManager.newSequentialScheduledExecutorService("unit-test",
+                "thread-1");
+
+        AtomicBoolean done = new AtomicBoolean(false);
+
+        service.submit(() -> {
+            Thread.sleep(1000);
```

Comment:
Could we reduce the waiting to `100` throughout the tests or does that make them instable? I would try to reduce the execution time as much as possible.

Reply:
100ms should be more than enought